### PR TITLE
[Doc] add query hints

### PR DESCRIPTION
**Pull request by hellolilyliuyi**
Fixes #issue

## What type of PR is this:
- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [x] Doc
- [ ] Tool

## Checklist:
- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4

### CODE REVIEW

File: docs/administration/Query_planning.md [hunk 1]
```diff
@@ -193,6 +193,104 @@ By removing the specific expressions (only keep the operators), the query plan c
 
 ![8-5](../assets/8-5.png)
 
+## Query hint
+
+Query hints are directives or comments that explicitly suggest the query optimizer on how to execute a query. Hints only take effect within a single query.
+
+### System variables hint
```
Comment (amber-create) [inline docs/administration/Query_planning.md:200]: image
remove "s" in "variables"

File: docs/administration/Query_planning.md [hunk 2]
```diff
@@ -193,6 +193,104 @@ By removing the specific expressions (only keep the operators), the query plan c
...
+
+You can set one or more system variable hints by using the `/*+ ... */` comment syntax in SELECT and SUBMIT TASK statements. You can also use system variable hints in SELECT clauses that included in other statements such as CREATE MATERIALIZED VIEW AS SELECT and CREATE VIEW AS SELECT.
```
Comment (amber-create) [inline docs/administration/Query_planning.md:202]: image
The two sentences can be combined into one: You can set one or more system variable hints by using the `/*+ ... */` comment syntax in SELECT and SUBMIT TASK statements, or in SELECT clauses that are (NOTE that "are" should be added between "that" and "included") included in other statements such as CREATE MATERIALIZED VIEW AS SELECT and CREATE VIEW AS SELECT.

File: docs/administration/Query_planning.md [hunk 3]
```diff
@@ -193,6 +193,104 @@ By removing the specific expressions (only keep the operators), the query plan c
...
+PARTITION BY dt 
+DISTRIBUTED BY HASH(`key`) 
+BUCKETS 10 
+REFRESH ASYNC 
+AS SELECT /*+ SET_VAR(query_timeout=500) */ * from dual;
+~~~
+
+### Join hint
+
+For multi-table join queries, the optimizer usually automatically selects the optimal join execution method. In special cases, you can use join hint to explicitly suggest the join execution method to the optimizer or disable Join Reorder. Currently, join hint supports suggesting Shuffle Join, Broadcast Join, Bucket Shuffle Join, and Colocate Join as a join execution method. When join hint is used, the optimizer does not perform Join Reorder. So you need to select the smaller table as the right table. Additionally, when suggesting Colocate Join or Bucket Shuffle Join as the join execution method, make sure that the data distribution of the joined table meets the requirements of these join execution methods. Otherwise, the suggested join execution method cannot take effect.
```
Comment (amber-create) [inline docs/administration/Query_planning.md:238]: image
"automatically" can be removed

Comment (amber-create) [inline docs/administration/Query_planning.md:238]: image
"join hint" is countable. It's better to say "a join hint" or "join hints".

Comment (amber-create) [inline docs/administration/Query_planning.md:238]: image
should be "or" I think

File: docs/administration/Query_planning.md [hunk 4]
```diff
@@ -193,6 +193,104 @@ By removing the specific expressions (only keep the operators), the query plan c
...
+
+  If you need to shuffle the data rows with the same bucket key values from tables A and B onto the same machine before a Join operation is performed, you can hint the join execution method as Shuffle Join.
+
+  ~~~SQL
+  select k1 from t1 join [SHUFFLE] t2 on t1.k1 = t2.k2 group by t2.k2;
+  ~~~
+
+- Broadcast Join
+  
+  If table A is a large table and table B is a small table, you can hint the join execution method as Broadcast Join. The data of the small table B is fully broadcasted to the machines on which the data of table A resides, and then the Join operation is performed. Compared to Shuffle Join, Broadcast Join saves the cost of shuffling the data of table A.
```
Comment (amber-create) [inline docs/administration/Query_planning.md:262]: image
consider removing "small"

File: docs/administration/Query_planning.md [hunk 5]
```diff
@@ -193,6 +193,104 @@ By removing the specific expressions (only keep the operators), the query plan c
...
+  
+  If tables A and B belong to the same Colocation Group which is specified during table creation, the data rows with the same bucket key values from tables A and B are distributed on the same BE node. When the Join equijoin expression contains the bucketing key of tables A and B in the join query, you can hint the join execution method as Colocate Join. Data with the same key values are directly joined locally, reducing the time spent on data transmission between nodes and improving query performance.
+
+  ~~~SQL
+  select k1 from t1 join [COLOCATE] t2 on t1.k1 = t2.k2 group by t2.k2;
+  ~~~
+
+### View join execution method
+
+Use the `EXPLAIN` command to view the actual join execution method**.** If the returned result shows that the join execution method matches the join hint, it means that the join hint is effective.
```
Comment (amber-create) [inline docs/administration/Query_planning.md:286]: image
improper format

File: docs/administration/Query_planning.md [hunk 6]
```diff
@@ -193,6 +193,104 @@ By removing the specific expressions (only keep the operators), the query plan c
...
+
+> **NOTE**
+>
+> Join Hint is case-insensitive.
+
+#### Examples
+
+- Shuffle Join
+
+  If you need to shuffle the data rows with the same bucket key values from tables A and B onto the same machine before a Join operation is performed, you can hint the join execution method as Shuffle Join.
```
Comment (amber-create) [inline docs/administration/Query_planning.md:254]: image
 should be "bucketing key" I think. There are two sentences with "bucket key" in this article.

**Review by amber-create — DISMISSED** [review]

File: docs/administration/Query_planning.md [hunk 7]
```diff
@@ -193,6 +193,104 @@ By removing the specific expressions (only keep the operators), the query plan c
 
 ![8-5](../assets/8-5.png)
 
+## Query hint
+
+Query hints are directives or comments that explicitly suggest the query optimizer on how to execute a query. Hints only take effect within a single query.
+
+### System variable hint
+
+You can set one or more system variable hints by using the `/*+ ... */` comment syntax in SELECT and SUBMIT TASK statements, or in SELECT clauses that are included in other statements such as CREATE MATERIALIZED VIEW AS SELECT and CREATE VIEW AS SELECT.
```
Comment (evelynzhaojie) [inline docs/administration/Query_planning.md:202]: add link to system variable

File: docs/administration/Query_planning.md [hunk 8]
```diff
@@ -193,6 +193,104 @@ By removing the specific expressions (only keep the operators), the query plan c
...
+PARTITION BY dt 
+DISTRIBUTED BY HASH(`key`) 
+BUCKETS 10 
+REFRESH ASYNC 
+AS SELECT /*+ SET_VAR(query_timeout=500) */ * from dual;
+~~~
+
+### Join hint
+
+For multi-table join queries, the optimizer usually selects the optimal join execution method. In special cases, you can use a join hint to explicitly suggest the join execution method to the optimizer or disable Join Reorder. Currently, a join hint supports suggesting Shuffle Join, Broadcast Join, Bucket Shuffle Join, or Colocate Join as a join execution method. When a join hint is used, the optimizer does not perform Join Reorder. So you need to select the smaller table as the right table. Additionally, when suggesting Colocate Join or Bucket Shuffle Join as the join execution method, make sure that the data distribution of the joined table meets the requirements of these join execution methods. Otherwise, the suggested join execution method cannot take effect.
```
Comment (evelynzhaojie) [inline docs/administration/Query_planning.md:238]: add link for colocate join

File: docs/administration/Query_planning.md [hunk 9]
```diff
@@ -193,6 +193,104 @@ By removing the specific expressions (only keep the operators), the query plan c
 
 ![8-5](../assets/8-5.png)
 
+## Query hint
+
+Query hints are directives or comments that explicitly suggest the query optimizer on how to execute a query. Hints only take effect within a single query.
```
Comment (evelynzhaojie) [inline docs/administration/Query_planning.md:198]: align with chinese version

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Review by amber-create — APPROVED** [review]

**Review by evelynzhaojie — APPROVED** [review]

**Comment by hellolilyliuyi** [comment]
@mergify backport branch-3.1 branch-3.0 branch-2.5

**Pull request merged**